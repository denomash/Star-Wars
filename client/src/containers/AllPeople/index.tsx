import React, { FC } from "react";
import { navigate } from "@reach/router";

// types
import { PageType } from "../../types/PageType";
import { IPerson } from "../../types/common";

import PersonList from "../../components/PersonList/PersonList";
import Pagination from "../../components/Pagination/Pagination";
import Loader from "../../components/Common/Loader";

import { PeopleConsumer } from "../../contexts/PeopleProvider";

const AllPeople: FC<PageType> = () => {
  return (
    <div className="container">
      <div className="center">
        <h1>The Star Wars People</h1>
      </div>

      <PeopleConsumer>
        {({ loading, people, count }: any) => {
          return (
            <>
              <div className="center">
                <Pagination
                  next={people?.next}
                  previous={people?.previous}
                  count={count}
                />
              </div>

              {loading ? (
                <div className="loader">
                  <Loader />
                </div>
              ) : (
                <div className="people-container">
                  {people?.people.map((person: IPerson) => (
                    <div onClick={() => navigate(`/details/${person.name}`)}>
                      <PersonList key={person.name} person={person} />
                    </div>
                  ))}
                </div>
              )}
              <div className="center">
                <Pagination
                  next={people?.next}
                  previous={people?.previous}
                  count={count}
                />
              </div>
            </>
          );
        }}
      </PeopleConsumer>
    </div>
  );
};

export default AllPeople;
