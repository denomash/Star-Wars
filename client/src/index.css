/* Selects each and every element in the page */
*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

body {
  font-family: "Open Sans", sans-serif;
  background: #f2f2f2;
  color: #101010;
}
