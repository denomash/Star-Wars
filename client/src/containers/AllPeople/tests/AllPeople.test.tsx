import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import TestRenderer from "react-test-renderer";

import { MockedProvider } from "@apollo/client/testing";
import { createHistory, LocationProvider } from "@reach/router";

import AllPeople from "../index";

const renderer = new ShallowRenderer();
const mocks: Array<any> = [];
let history = createHistory(window);

describe("<AllPeople />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<AllPeople />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("renders without error", () => {
    const Component = () => (
      <LocationProvider history={history}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <AllPeople />
        </MockedProvider>
      </LocationProvider>
    );
    const component = TestRenderer.create(<Component />);

    const tree: any = component.toJSON();
    expect(tree.children[0].children[0].children[0]).toContain("The Star Wars People");
  });
});
