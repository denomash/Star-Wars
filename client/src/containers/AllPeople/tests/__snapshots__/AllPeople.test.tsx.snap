// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllPeople /> should render and match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="center"
  >
    <h1>
      The Star Wars People
    </h1>
  </div>
  <Context.Consumer>
    [Function]
  </Context.Consumer>
</div>
`;
